moabb/tests/test_splits.py

```diff
@@ -74,13 +103,71 @@ def test_is_shuffling():
-    split = WithinSessionSplitter(cv_class=TimeSeriesSplit, max_train_size=2)
+    split = splitter(cv_class=TimeSeriesSplit, max_train_size=2)
 
     for train, test in split.split(y, metadata):
         # Check if the output is the same as the input
-        assert len(train) == 2
-        assert len(test) == 20
+        assert len(train) <= 2  # Due to TimeSeriesSplit constraints
+        assert len(test) >= 20
+
+
+@pytest.mark.parametrize("shuffle, random_state", [(True, 0), (True, 42), (False, None)])
+def test_cross_session(shuffle, random_state):
+    X, y, metadata = paradigm.get_data(dataset=dataset)
+
+    split = CrossSessionSplitter(shuffle=shuffle, random_state=random_state)
+
+    for idx_train_splitter, idx_test_splitter in split.split(y, metadata):
+        # Check if the output is the same as the input
+        session_train = metadata.iloc[idx_train_splitter]["session"].unique()
+        session_test = metadata.iloc[idx_test_splitter]["session"].unique()
+        assert not np.intersect1d(session_train, session_test).size
+        assert (
+            np.union1d(session_train, session_test).size
+            == metadata["session"].unique().size
+        )
+
+
+@pytest.mark.parametrize("shuffle, random_state", [(True, 0), (True, 42), (False, None)])
+def test_cross_session_compatibility(shuffle, random_state):
+    _, y, metadata = paradigm.get_data(dataset=dataset)
+
+    splitter = CrossSessionSplitter(shuffle=shuffle, random_state=random_state)
+
+    for (idx_train, idx_test), (idx_train_splitter, idx_test_splitter) in zip(
+        eval_split_cross_session(shuffle=shuffle, random_state=random_state),
+        splitter.split(y, metadata),
+    ):
+        assert np.array_equal(idx_train, idx_train_splitter)
+        assert np.array_equal(idx_test, idx_test_splitter)
+
+
+def test_cross_session_is_shuffling():
+    X, y, metadata = paradigm.get_data(dataset=dataset)
+
+    splitter_no_shuffle = CrossSessionSplitter(shuffle=False)
+    splitter_shuffle = CrossSessionSplitter(shuffle=True, random_state=3)
+
+    splits_no_shuffle = list(splitter_no_shuffle.split(y, metadata))
+    splits_shuffle = list(splitter_shuffle.split(y, metadata))
+
+    for i, ((train_ns, test_ns), (train_s, test_s)) in enumerate(
+        zip(splits_no_shuffle, splits_shuffle)
+    ):
+        print(f"\nFold {i}:")
+        print(f"Train no shuffle sessions: {metadata.iloc[train_ns]['session'].unique()}")
+        print(f"Test no shuffle sessions : {metadata.iloc[test_ns]['session'].unique()}")
+        print(f"Train shuffled sessions  : {metadata.iloc[train_s]['session'].unique()}")
+        print(f"Test shuffle sessions    : {metadata.iloc[test_s]['session'].unique()}")
+
+        assert not np.array_equal(
+            train_ns, train_s
+        ), f"Fold {i} train indices identical despite shuffle"
+        assert not np.array_equal(
+            test_ns, test_s
+        ), f"Fold {i} test indices identical despite shuffle"
```

Comment:
I would also check that if you have section 1, 2, 3, the order is not always the same.
Eventhough it is probably less important than for `WithinSession` where the shuffling of the data is more important than in this context with `LeaveOneGroupOut`.